tmt/cli.py

```diff
@@ -2030,20 +2028,30 @@ def setup_completion(shell: str, install: bool) -> None:
     else:
         script = Path(config.path) / f'{COMPLETE_SCRIPT}.{shell}'
 
-    # SIM115: Use context handler for opening files. Would not reduce complexity here.
-    out = open(script, 'w') if install else sys.stdout  # noqa: SIM115
-    subprocess.run(f'{COMPLETE_VARIABLE}={shell}_source tmt',
-                   shell=True, stdout=out)
+    env_var = {COMPLETE_VARIABLE: f'{shell}_source'}
+
+    logger = context.obj.logger
+
+    completions = Command('tmt').run(env=tmt.utils.Environment.from_dict(env_var),
+                                     cwd=None,
+                                     logger=context.obj.logger
+                                     ).stdout
+    if not completions:
+        logger.warning("Unable to generate shell completion")
+        return
 
     if install:
-        out.close()
+        Path(script).write_text(str(completions))
```

Comment:
```suggestion
        assert completions is not None
        Path(script).write_text(completions)
```

How about doing type narrowing (probably a bit higher up) instead of the `str()`

Reply:
damn, I left it there by accident. Thanks! You guys are so good at reviewing...   
The assert is not needed, as there is a check higher up.